### PR TITLE
[doc] Improve manifest documentation links

### PR DESCRIPTION
**Pull request by dg0yt**
This PR fixes a broken documentation link, and it improves the TOC on the front page.

I got lost in the documentation a few times. Maybe this needs even more explicit language, e.g. "user manifest files" and "port manifest files".

### CODE REVIEW

File: docs/maintainers/control-files.md [hunk 1]
```diff
@@ -1,7 +1,7 @@
 # CONTROL files
 
 **CONTROL files are retained for backwards compatibility with earlier versions of vcpkg;
-all new features are added only to [vcpkg.json](manifests.md), and we recommend using vcpkg.json for any newly authored port.**
+all new features are added only to [vcpkg.json manifest files](manifest-files.md), and we recommend using vcpkg.json for any newly authored port.**
```
Comment (NancyLi1013) [inline docs/maintainers/control-files.md:4]:
```suggestion
all new features are added only to [vcpkg.json](manifest-files.md), and we recommend using vcpkg.json for any newly authored port.**
```
`manifest files` seems not needed here. Or maybe updated it as  `Manifest files - vcpkg.json`?

**Review by NancyLi1013 — CHANGES_REQUESTED** [review]

File: docs/maintainers/control-files.md (hunk 1 above)
Comment (dg0yt) [reply docs/maintainers/control-files.md:4]: IMHO "vcpkg.json manifest files" is what reads most fluently (apart form simply "manifest files"). "Manifest files - vcpkg.json" is fine for headlines and toc, but not in plain text.
Disclaimer: I'm not a native speaker.

But I can role back to plain "vcpkg.json".

Comment (NancyLi1013) [reply docs/maintainers/control-files.md:4]: Thanks for your description about this point. Sounds good.

**Review by NancyLi1013 — APPROVED** [review]

**Review by ras0219-msft — APPROVED** [review]
Thanks for the PR!

**Comment by dg0yt** [comment]
This change doesn't seem to arrive at https://vcpkg.readthedocs.io/en/latest/maintainers/control-files/

**Pull request merged**